File: fixtures/l1/js/main.js
// A tiny fixture: one function with a discount branch and a guard.
function applyTiered(total, tier) {
  const base = total
  if (tier >= 2) {
    const bonus = tier * 0.05
    return base * (1 - bonus)
  }
  if (total < 0) {
    throw new Error('total must not be negative')
  }
  return base
}

console.log('result', applyTiered(100, 2))

File: fixtures/l1/go/main.go
// A tiny fixture: one function with a discount branch and a guard.
package main

import "fmt"

func applyTiered(total float64, tier int) float64 {
	base := total
	if tier >= 2 {
		bonus := float64(tier) * 0.05
		return base * (1 - bonus)
	}
	if total < 0 {
		panic("total must not be negative")
	}
	return base
}

func main() {
	fmt.Println("result", applyTiered(100, 2))
}
